### PR TITLE
Make CSS2 meet pubrules

### LINKED ISSUE #2560: Fix pubrules failures in CSS2
Many of these things are just changes since we last published (like preferring HTTPS URLs), new host organisations, style sheet updates, etc. We also need to move the document to valid HTML5.

### PR DESCRIPTION
**Pull request by gsnedders**
Fixes #2560.

This PR can kinda be summarised as a PR of four parts:

 1. Replacing `<a name>` with `<span id>` (yes, I know, it's ugly)
 2. Making the pages valid HTML LS
 3. Updating to the current stylesheet
 4. Fixing the cover page for pubrules

Suggest reviewing this commit-by-commit, given the total diff here is massive. You probably mostly want to ignore changes in css2/*.html until you've checked the rest (in the build system / source files), then check the changes seem approximately right.

### CODE REVIEW

**Comment by svgeesus** [comment]
Uh wait, I was planning to approve these one by one!

**Comment by gsnedders** [comment]
FWIW, I'd appreciate this not hanging around forever given the first bunch (moving to `<span id>`) cause conflicts all over the place!

**Comment by gsnedders** [comment]
Also: https://deploy-preview-1--unruffled-galileo-5a0f9a.netlify.com/cover.html has this.

File: css2/bin/addianch [hunk 1]
```diff
@@ -16,6 +16,7 @@ sub cleanup {
     my $h = $_[0];
     $h =~ tr/\n\t/ /s;		# newline & tab -> spaces, and collapse
     $h =~ s/<\/?a\b.*?>//gio;	# remove A tags
+    $h =~ s/<\/?span\b.*?>//gio;	# remove A tags
```
Comment (frivoal) [inline css2/bin/addianch]: The comment on this line is not accurate. Copy&paste leftover?

**Comment by frivoal** [comment]
Caveat about this review: This is a large PR, and I went through it relatively quickly. It is quite likely that I missed some things. With that said, overall it looks good, and here are my detailed commit-per-commit comments:

* [First commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/8c9c7ab16d6bb8787dbcf6d933c5d0df6667b7a1) seems ok, except that some `propinst-*` classes got dropped in the build output (e.g. `propinst-pause-before` in aural.html). I am not sure that they served a purpose, so maybe it's OK that they're gone, but if they did something useful, it would be good to figure out what killed them and to bring them back.
* [Second commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/bef9f1cd3938b9a240dfa8010f5fe919a2bd13f2) OK
* [Third commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/0eaa16d4865139926782e47adc4e1f19b83324c0) OK (should have cought it in first commit review, but didn't. Sorry)
* [Fourth commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/2ce6e0ff2854961a0996b49551957cd1c9d7f387) OK
* [5th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/8f819d91127cc5a58c41110015aa3cc530e97425) OK
* [6th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/d12c8ecd1dec7c8c99e848bce3642ac0b7ff34e2) OK, except for one small mistake in a comment, see https://github.com/w3c/csswg-drafts/pull/4850#pullrequestreview-372409072
* [7th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/e13136fd4891eaaddabaf39b0775f26470a5d02d) somewhat subjective, but OK
* [8th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/e01302f038a3a7b70adf86d7d6ad022562834bc5) The statement that no UA would do anything with bare links is not correct. For instance, presto-based opera would show the `rel=first` and `rel=index` links in the info panel. While that UA is no longer relevant, there may be others. That said, I am not sure this will cause issues for anyone in practice. But do we solve anything by removing them?
* [9th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/519bc2b9a8b6cda75149a33f471b56956bea0f39) OK
* [10th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/2e47108a037394e8d238d15f47b4095d43bbb1fd) OK
* [11th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/38183e52e61244dc1e2db3b365885215fa6d54c2) OK
* [12th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/cb074cdb00041f0059a365ff12ef23d2ad947f43) It seems fine, but I am not sure what the benefit of switching from `<pre><code>…</code></pre>` to `<pre>…</pre>` is. As far as I can tell, bikeshed happily accepts both. Not a blocker, but I'm curious.
* [13th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/21aee42aead5d256033ec088ce86702324e5a482) OK
* [14th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/11c4ba4adc6ef1aec80554badef6d2102198de3c) OK
* [15th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/c608c11ec64d70a81ed5ec8533c468cc69ae57d8) Not a blocker, but seems unfortunate, this was useful. I wonder if it can be preserved in a different form.

**Comment by gsnedders** [comment]
> But do we solve anything by removing them?

They're validation errors as deprecated link forms.

> Caveat about this review: This is a large PR, and I went through it relatively quickly. It is quite likely that I missed some things. With that said, overall it looks good, and here are my detailed commit-per-commit comments:
> 
> * [First commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/8c9c7ab16d6bb8787dbcf6d933c5d0df6667b7a1) seems ok, except that some `propinst-*` classes got dropped in the build output (e.g. `propinst-pause-before` in aural.html). I am not sure that they served a purpose, so maybe it's OK that they're gone, but if they did something useful, it would be good to figure out what killed them and to bring them back.

Oh, uh, that I missed. That I should look into at least, because it makes me doubt correctness of the whole thing.

> * [Second commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/bef9f1cd3938b9a240dfa8010f5fe919a2bd13f2) OK
> * [Third commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/0eaa16d4865139926782e47adc4e1f19b83324c0) OK (should have cought it in first commit review, but didn't. Sorry)
> * [Fourth commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/2ce6e0ff2854961a0996b49551957cd1c9d7f387) OK
> * [5th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/8f819d91127cc5a58c41110015aa3cc530e97425) OK
> * [6th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/d12c8ecd1dec7c8c99e848bce3642ac0b7ff34e2) OK, except for one small mistake in a comment, see [#4850 (review)](https://github.com/w3c/csswg-drafts/pull/4850#pullrequestreview-372409072)
> * [7th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/e13136fd4891eaaddabaf39b0775f26470a5d02d) somewhat subjective, but OK

Note that this is very different now the content has a max-width of 50em, rather than being the entire width of the viewport. At the current width and font-size, it's not very readable. That said very much the goal just to get this publishable, rather than perfect stylistically!

> * [8th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/e01302f038a3a7b70adf86d7d6ad022562834bc5) The statement that no UA would do anything with bare links is not correct. For instance, presto-based opera would show the `rel=first` and `rel=index` links in the info panel. While that UA is no longer relevant, there may be others. That said, I am not sure this will cause issues for anyone in practice. But do we solve anything by removing them?

AFAIK, no current UA does anything with them. We need need to avoid emitting `CSS-properties` there (which is a validation error as an undefined relation), however.

> * [9th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/519bc2b9a8b6cda75149a33f471b56956bea0f39) OK
> * [10th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/2e47108a037394e8d238d15f47b4095d43bbb1fd) OK
> * [11th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/38183e52e61244dc1e2db3b365885215fa6d54c2) OK
> * [12th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/cb074cdb00041f0059a365ff12ef23d2ad947f43) It seems fine, but I am not sure what the benefit of switching from `<pre><code>…</code></pre>` to `<pre>…</pre>` is. As far as I can tell, bikeshed happily accepts both. Not a blocker, but I'm curious.

Good question. I remember seeing something weird happening to the styling with it nested somewhere, but I can't reproduce this now.

> * [13th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/21aee42aead5d256033ec088ce86702324e5a482) OK
> * [14th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/11c4ba4adc6ef1aec80554badef6d2102198de3c) OK
> * [15th commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/c608c11ec64d70a81ed5ec8533c468cc69ae57d8) Not a blocker, but seems unfortunate, this was useful. I wonder if it can be preserved in a different form.

Again, more concerned about getting this publishable than perfect. If we want a better TOC for long documents we should perhaps consider doing this more generically for W3C specs.

**Comment by gsnedders** [comment]
> Oh, uh, that I missed. That I should look into at least, because it makes me doubt correctness of the whole thing.

Oh, from the TOC. Not clear that cross-refs have so much value there, given we can't actually turn them into links anyway?

**Comment by frivoal** [comment]
>>> * [First commit](https://github.com/w3c/csswg-drafts/pull/4850/commits/8c9c7ab16d6bb8787dbcf6d933c5d0df6667b7a1) seems ok, except that some `propinst-*` classes got dropped in the build output (e.g. `propinst-pause-before` in aural.html). I am not sure that they served a purpose, so maybe it's OK that they're gone, but if they did something useful, it would be good to figure out what killed them and to bring them back.

>>  Oh, uh, that I missed. That I should look into at least, because it makes me doubt correctness of the whole thing.

> Oh, from the TOC. Not clear that cross-refs have so much value there, given we can't actually turn them into links anyway?

Doesn't seem useful to me either. It doesn't seem to me that it's a problem that these have been dropped, but since I don't understand why they were there in the first place, I may be missing something.

> That said very much the goal just to get this publishable, rather than perfect stylistically!

Agreed. I raised what I found, but none of this should be considered a blocker.

**Comment by gsnedders** [comment]
> Good question. I remember seeing something weird happening to the styling with it nested somewhere, but I can't reproduce this now.

Ah, it's `<pre><code class="css">`, because of the styles on `.css` in Bikeshed.

**Pull request merged**